search-commons/src/main/java/no/unit/nva/search/models/SearchResponseDto.java

```diff
@@ -184,17 +171,65 @@ public boolean equals(Object o) {
 
         SearchResponseDto that = (SearchResponseDto) o;
         return processingTime == that.processingTime
-                && size == that.size
-                && Objects.equals(context, that.context)
-                && Objects.equals(id, that.id)
-                && Objects.equals(hits, that.hits);
+               && size == that.size
+               && Objects.equals(context, that.context)
+               && Objects.equals(id, that.id)
+               && Objects.equals(hits, that.hits);
     }
 
     private static List<JsonNode> extractSourcesList(SearchResponse searchResponse) {
         return Arrays.stream(searchResponse.getHits().getHits())
-                .map(SearchHit::getSourceAsMap)
-                .map(source -> objectMapperWithEmpty.convertValue(source, JsonNode.class))
-                .collect(Collectors.toList());
+            .map(SearchHit::getSourceAsMap)
+            .map(source -> objectMapperWithEmpty.convertValue(source, JsonNode.class))
+            .collect(Collectors.toList());
+    }
+
+    private static JsonNode extractAggregations(SearchResponse searchResponse) {
+        JsonNode json = attempt(() -> objectMapperWithEmpty.readTree(searchResponse.toString())).orElseThrow();
+
+        JsonNode aggregations = json.get("aggregations");
+
+        if (aggregations == null) {
+            return null;
+        }
+
+        var jsonWithCorrectNames = stripAggregationName(aggregations);
+
+        return aggregationFieldsToCamelCase(jsonWithCorrectNames);
+    }
+
+    private static ObjectNode stripAggregationName(JsonNode aggregations) {
+        var outputAggregationNode = objectMapperWithEmpty.createObjectNode();
+
+        var iterator = aggregations.fields();
+        while (iterator.hasNext()) {
+            var child = iterator.next();
+            var newName = child.getKey().replaceFirst(WORD_ENDING_WITH_HASHTAG_REGEX, "");
+            outputAggregationNode.set(newName, child.getValue());
+        }
+
+        return outputAggregationNode;
+    }
+
+    private static ObjectNode aggregationFieldsToCamelCase(JsonNode aggregations) {
+        var json = (ObjectNode) aggregations;
+
+        for (JsonNode aggregationNode : json) {
+            var aggregation = (ObjectNode) aggregationNode;
+            AGGREGATION_FIELD_TO_CHANGE.forEach((oldname, newName) -> renameField(aggregation, oldname, newName));
+
+            for (JsonNode bucketNode : aggregation.get(BUCKETS_FIELD_NAME)) {
+                var bucket = (ObjectNode) bucketNode;
+                BUCKET_FIELD_TO_CHANGE.forEach((oldName, newName) -> renameField(bucket, oldName, newName));
```

Comment:
What happens if the field does not exist? My guess is that it will be added with null data. Is this the wanted behaviour?